quic/s2n-quic-rustls/src/cipher_suite.rs

```diff
@@ -11,9 +12,18 @@ use s2n_quic_core::crypto::{self, packet_protection, scatter, tls, HeaderProtect
 /// `aws_lc_rs` is the default crypto provider since that is also the
 /// default used by rustls.
 pub(crate) fn default_crypto_provider() -> Result<CryptoProvider, rustls::Error> {
+    cfg_if!(
+        if #[cfg(feature = "fips")] {
+            let crypto = rustls::crypto::default_fips_provider();
+            assert!(crypto.fips());
```

Comment:
is this actually necessary? From reading the [rustls docs](https://docs.rs/rustls/latest/rustls/crypto/struct.CryptoProvider.html), seems like not?

> [crypto::aws_lc_rs::default_provider](https://docs.rs/rustls/latest/rustls/crypto/aws_lc_rs/fn.default_provider.html): (behind the aws_lc_rs feature, which is enabled by default). This provider uses the [aws-lc-rs](https://github.com/aws/aws-lc-rs) crate. The fips crate feature makes this option use FIPS140-3-approved cryptography.

Reply:
Great question, no it's not.

By using the explicit `default_fips_provider()` method we avoid assumptions about any internal rustls details, which I feel is a bit more resilient. For example, rustls could theoretically change their fips crypto provider (openssl-fips) and use `aws-lc-sys` rather than `aws-lc-fips-sys` in the future. If that happens, we would no longer be fips compliant.

One possible alternative is:

```
let crypto = aws_lc_rs::default_provider();

#[cfg(feature = "fips")]
assert!(crypto.fips());
```

Reply:
I like the alternative. Its more explicit that we are always using `aws-lc`